x-pack/plugins/observability/public/pages/slo_details/components/header_control.tsx

```diff
@@ -37,43 +43,118 @@ export function HeaderControl({ isLoading, slo }: Props) {
     }
   };
 
+  const onCloseRuleFlyout = useCallback(
```

Comment:
Using a `useCallback` when setting local state is not necessary as the identity of the state function will be identical across renders. 
I would just do  

```
const handleCloseRuleFlyout = () => {
   setRuleFlyoutVisibility(false);
}
```

Reply:
honestly, I just copied the examples from triggerActionsUi :) Will update without hte useCallback